### DISCUSSION TITLE
Detecting Ruffle in AS2

### DISCUSSION THREAD

**Discussion by rikkoooo** [ruffle-rs/ruffle#15547] [Q&A]
How can i detect if my SWF movie is running with Ruffle in AS2?

**Comment by torokati44** [comment]
See: https://github.com/ruffle-rs/ruffle/issues/14555 :)

**Reply by rikkoooo** [reply]
I've opened this discussion after @n0samu says "Feel free to open a discussion if you want details on that.".

**Reply by torokati44** [reply]
Sure, there isn't a problem with that, I was just trying to show you some existing records about this.

**Reply by torokati44** [reply]
Oh, that was also by you... Sorry, didn't notice! 🤦‍♂️

**Reply by Lord-McSweeney** [reply]
If you can enable `allowscriptaccess` for the AS2 SWF you could test if `RufflePlayer` is defined on `window` (though that might not always work).
